keps/sig-network/2091-admin-network-policy/README.md

```diff
@@ -0,0 +1,1116 @@
+selected by the AdminNetworkPolicy, as opposed to what NetworkPolicy rules imply.
+
+- Pass: Traffic that matches a `Pass` rule will skip all further rules from all 
+  ANPs and instead be enforced by the K8s NetworkPolicies. If there is no K8s 
+  NetworkPolicy rule match, traffic will be allowed.
+- Deny: Traffic that matches a `Deny` rule will be dropped. 
+- Allow: Traffic that matches an `Allow` rule will be allowed.
+
+AdminNetworkPolicy `Deny` rules are useful for administrators to explicitly
+block traffic with malicious clients, or workloads that poses security risks.
+Those traffic restrictions can only be lifted once the `Deny` rules are deleted
+or modified by the admin.
+
+On the other hand, the `Allow` rules can be used to call out traffic in the cluster
+that needs to be allowed for certain components to work as expected (egress to
+CoreDNS for example). Those traffic should not be blocked when developers apply
+NetworkPolicy to their Namespaces which isolates the workloads.
+
+AdminNetworkPolicy `Pass` rules allows an admin to delegate security posture for
+certain traffic to the Namespace owners by overriding any lower priority Allow or 
+Deny rules. For example, allowing intra-tenant traffic can be delegated to tenant 
+admins explicitly by the cluster admin with the use of `Pass` rules.
+
+#### Priority
+
+The policy instances will be ordered based on the numeric priority assigned to the
+ANP. A larger number corresponds to a higher precedence. For example, ANP with `priority`
+set to "1" will be considered as the policy with the lowest precedence, while "999" will
+be considered as highest precedence. Any positive priority, however, will have
+higher precedence over the namespaced NetworkPolicy instances in the cluster.
+
+Additionally, the special priority "0" can be used in the priority field to indicate
+that the rules in that policy instance shall be created at a priority lower than the
+Namespaced NetworkPolicies. 
+
+The relative precedence of the rules within a ANP will be determined by the order in which the
+rule is written. Thus, a rule that appears at the top of the ingress/egress rules would take the
+highest precedence.
+
+Conflict resolution: Two policies are considered to be conflicting if they are assigned the
+same `priority` and apply to the same resources or a union of resources. In order to avoid such conflicts, 
+we propose to include tooling for ANP resources to help alert the admin to potentially ambiguous ANP
+priority scenarios, more details in [risks and mitigations](#risks-and-mitigations). However, ultimately
+it will be the job of the CNI to decide how to handle overlapping priority situations in the respective 
+implementations. 
+
+For conformance and portability purposes, it will be required that implementations
+support the range of 1000 numeric ANP priorities (0 through 999) per cluster and a maximum of 100 rules
```

Comment:
So I specified this in the spec comments, but probably should have done it somewhere else in the KEP as well 

```
type AdminNetworkPolicySpec struct {
  // Int value bound to 0 - 999, with 0 signaling the special "baseline" priority
  Priority      Int 
  // Allows for multiple AdminNetworkPolicy rules at a single priority level
  // If empty no rules are applied
  // A total of 100 total rules will be allowed per each network policy instance, 
  // this rule count will be calculated as the total summation of the 
  // AdminNetworkPolicyRules, AdminNetworkPolicyIngressRules, and AdminNetworkPolicyEgressRules
  // in a single AdminNetworkPolicy Instance. 
  PolicyRules   []AdminNetworkPolicyRule
}
```

I think the goal of this is to prevent insanity, so that Admins create "just complex enough" ANPs and can't shoot themselves in the foot